wellregistry/wellregistry/templates/admin/bulk_upload.html

```diff
@@ -2,42 +2,55 @@
 {% load static %}
 
 {% block breadcrumbs %}
-    <div class="breadcrumbs">
-      <a href="{% url 'admin:index' %}">Home</a>
-        &rsaquo; <a href="{% url 'admin:registry_monitoringlocation_changelist' %}">Monitoring Location</a>
-        &rsaquo; Bulk Upload
-    </div>
+<div class="breadcrumbs">
+    <a href="{% url 'admin:index' %}">Home</a>
+    &rsaquo; <a href="{% url 'admin:registry_monitoringlocation_changelist' %}">Monitoring Location</a>
+    &rsaquo; Bulk Upload
+</div>
 {% endblock %}
 
 
 {% block content %}
-    <p>To get the bulk upload template click <a href="{%  url 'admin:bulk_upload_template' %}">here</a></p>
-    <p>Note: A user can only upload a CSV file</p>
-    {% if errors %}
+<p>To get the bulk upload template click <a href="{%  url 'admin:bulk_upload_template' %}">here</a></p>
+<p>Note: A user can only upload a CSV file</p>
+{% if errors %}
+<ul class="errorlist">
+    {% for row_index, error in errors %}
+    <li>In row {{row_index}}
+        <ol>
+            {% for key, value in error.items %}
+            <li>{{ key }}: {{ value }}</li>
+            {% endfor %}
+        </ol>
+    </li>
+    {% endfor %}
+</ul>
+{% endif %}
+{% if warnings %}
+<ul class="errorlist">
+    {% for row_index, warning in warnings %}
+    <li>In row {{row_index}}
+        <ol>
+            {% for key, value in warning.items %}
+            <li>{{ key }}: {{ value }}</li>
+            {% endfor %}
+        </ol>
+    </li>
+    {% endfor %}
+</ul>
```

Comment:
We discussed putting a breadcrumb back to monitoring locations "page" here at the end of the list.
What if it was the actual breadcrumbs structure repeated. They could go were they like.

Reply:
spacing fixed